File: Checks/PWR003/gravity.f90
module gravity_getters
contains
  ! Returns the gravity on planet `planetIndex` in m/s2
  pure function gravity_pure(planetIndex)
    use iso_c_binding, only: c_int, c_double
    
    implicit none
    integer(kind=c_int), intent(in) :: planetIndex
    real(kind=c_double) :: gravity_pure

    select case (planetIndex) 
      case (0) ! Mercury
        gravity_pure = 3.7
      case (1) ! Venus
        gravity_pure = 8.9
      case (2) ! Earth
        gravity_pure = 9.8
      case (3) ! Mars
        gravity_pure = 3.7
      case (4) ! Jupiter
        gravity_pure = 23.1
      case (5) ! Saturn
        gravity_pure = 9.0
      case (6) ! Uranus
        gravity_pure = 8.7
      case (7) ! Neptune
        gravity_pure = 11.0
      case default
        gravity_pure = 0.0
    end select
  end function gravity_pure
  
  function gravity_impure(planetIndex)
    use iso_c_binding, only: c_int, c_double
    
    implicit none
    integer(kind=c_int), intent(in) :: planetIndex
    real(kind=c_double) :: gravity_impure

    select case (planetIndex) 
      case (0) ! Mercury
        gravity_impure = 3.7
      case (1) ! Venus
        gravity_impure = 8.9
      case (2) ! Earth
        gravity_impure = 9.8
      case (3) ! Mars
        gravity_impure = 3.7
      case (4) ! Jupiter
        gravity_impure = 23.1
      case (5) ! Saturn
        gravity_impure = 9.0
      case (6) ! Uranus
        gravity_impure = 8.7
      case (7) ! Neptune
        gravity_impure = 11.0
      case default
        gravity_impure = 0.0
    end select
  end function gravity_impure
end module gravity_getters

File: Checks/PWR003/gravity.c
// Returns the gravity on planet `planetIdx` in m/s2
double gravity(const int planetIdx) {
  if (planetIdx == 0) // Mercury
    return 3.7;
  if (planetIdx == 1) // Venus
    return 8.9;
  if (planetIdx == 2) // Earth
    return 9.8;
  if (planetIdx == 3) // Mars
    return 3.7;
  if (planetIdx == 4) // Jupiter
    return 23.1;
  if (planetIdx == 5) // Saturn
    return 9.0;
  if (planetIdx == 6) // Uranus
    return 8.7;
  if (planetIdx == 7) // Neptune
    return 11.0;

  return 0.0;
}
